packages/core/src/Stores/client-store.js

```diff
@@ -1445,7 +1445,7 @@ export default class ClientStore extends BaseStore {
 
     get residence() {
         if (this.is_logged_in) {
-            return this.account_settings.country_code ?? '';
+            return this.accounts[this.loginid].country ?? '';
```

Comment:
here we set the residence on every authorize request which happens every time when the user landed on app.deriv.com/any_url or logins into the app